.github/workflows/main.yml

```diff
@@ -3,19 +3,36 @@ name: Main
 on: [push, pull_request]
 
 jobs:
+
   lint:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - name: Setup Python
-        uses: actions/setup-python@v1
+      - uses: actions/setup-python@v2
         with:
           python-version: 3.8
-      - name: Install tox
-        run: python -m pip install tox
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip setuptools wheel tox
       - name: Run linting
         run: python -m tox -e lint
 
+  types:
+    strategy:
+      matrix:
+        python: [3.6, 3.7, 3.8]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip setuptools wheel tox
```

Comment:
This "stanza" duplicated in all jobs, and I haven't looked into ways to DRY it out. There's an [open feature request](https://github.com/actions/setup-python/issues/77) on the `setup-python` action.

Reply:
There's also work in progress to support composite (e.g. reusable) actions:

- [Next Steps for Fully Functioning Composite Actions · Issue #646 · actions/runner](https://github.com/actions/runner/issues/646)
- [Creating a composite run steps action - GitHub Docs](https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-composite-run-steps-action)
- [Example using action in the same repository as the workflow - GitHub Docs](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-using-action-in-the-same-repository-as-the-workflow)

But it looks like we're stuck with this for now.